clang/lib/Basic/Sanitizers.cpp

```diff
@@ -43,6 +44,23 @@ std::optional<double> SanitizerMaskCutoffs::operator[](unsigned Kind) const {
 
 void SanitizerMaskCutoffs::clear(SanitizerMask K) { set(K, 0); }
 
+std::optional<std::vector<int>>
+SanitizerMaskCutoffs::getAllScaled(int ScalingFactor) const {
+  std::vector<int> scaledCutoffs;
+
+  bool anyNonZero = false;
+  for (unsigned int i = 0; i < SanitizerKind::SO_Count; ++i) {
```

Comment:
```
 bool AnyCutOff = false;
  for (unsigned int i = 0; i < SanitizerKind::SO_Count; ++i) {
   auto C = (*this)[i];
    scaledCutoffs.push_back(C.has_value() ? std::clamp(round(ScalingFactor * *C)) : 0);
    anyNonZero |= C.has_value();
  }
```

Reply:
Done, except for the clamping: what is the reasonable value to clamp to? Shouldn't the clamping, if any, be done in `operator[]`?

Reply:
Assuming we map [0, 1] -> [0, ScalingFactor], then ScalingFactor